File: src/test/java/net/flyingfishflash/loremlist/unit/domain/lrmlist/LrmListNotFoundTests.java
package net.flyingfishflash.loremlist.unit.domain.lrmlist;

import static org.assertj.core.api.Assertions.assertThat;

import java.util.Set;
import java.util.UUID;
import net.flyingfishflash.loremlist.domain.lrmlist.ListNotFoundException;
import org.junit.jupiter.api.Test;

class LrmListNotFoundTests {

  @Test
  void primaryConstructorDefaultMessage() {
    UUID id = UUID.randomUUID();
    ListNotFoundException exception = new ListNotFoundException(id);
    assertThat(exception.getMessage()).isEqualTo("List could not be found.");
  }

  @Test
  void primaryConstructorNonDefaultMessage() {
    ListNotFoundException exception = new ListNotFoundException(UUID.randomUUID(), "Lorem Ipsum");
    assertThat(exception.getMessage()).isEqualTo("Lorem Ipsum");
  }

  @Test
  void secondaryConstructorDefaultMessageSetSizeOne() {
    Set<UUID> ids = Set.of(UUID.randomUUID());
    ListNotFoundException exception = new ListNotFoundException(ids);
    assertThat(exception.getMessage()).isEqualTo("List could not be found.");
  }

  @Test
  void secondaryConstructorDefaultMessageSetSizeGreaterThanOne() {
    Set<UUID> ids = Set.of(UUID.randomUUID(), UUID.randomUUID());
    ListNotFoundException exception = new ListNotFoundException(ids);
    assertThat(exception.getMessage()).isEqualTo("Lists (2) could not be found.");
  }

  @Test
  void secondaryConstructorNonDefaultMessage() {
    ListNotFoundException exception =
        new ListNotFoundException(Set.of(UUID.randomUUID()), "Lorem Ipsum");
    assertThat(exception.getMessage()).isEqualTo("Lorem Ipsum");
  }
}

File: src/test/kotlin/net/flyingfishflash/loremlist/unit/domain/lrmlist/LrmListNotFoundTests.kt
package net.flyingfishflash.loremlist.unit.domain.lrmlist

import io.kotest.core.spec.style.DescribeSpec
import io.kotest.matchers.shouldBe
import net.flyingfishflash.loremlist.domain.lrmlist.ListNotFoundException
import java.util.UUID

class LrmListNotFoundTests :
  DescribeSpec({
    describe("ListNotFoundException()") {

      it("primary constructor: default message") {
        val id = UUID.randomUUID()
        val listNotFoundException = ListNotFoundException(id = id)
        listNotFoundException.message.shouldBe("List could not be found.")
      }

      it("primary constructor: non-default message") {
        val listNotFoundException = ListNotFoundException(id = UUID.randomUUID(), message = "Lorem Ipsum")
        listNotFoundException.message?.shouldBe("Lorem Ipsum")
      }

      it("secondary constructor: default message, set size = 1") {
        val ids = setOf(UUID.randomUUID())
        val listNotFoundException = ListNotFoundException(idCollection = ids)
        listNotFoundException.message.shouldBe("List could not be found.")
      }

      it("secondary constructor: default message, set size > 1") {
        val ids = setOf(UUID.randomUUID(), UUID.randomUUID())
        val listNotFoundException = ListNotFoundException(idCollection = ids)
        listNotFoundException.message.shouldBe("Lists (2) could not be found.")
      }

      it("secondary constructor: non-default message") {
        val listNotFoundException = ListNotFoundException(idCollection = setOf(UUID.randomUUID()), message = "Lorem Ipsum")
        listNotFoundException.message.shouldBe("Lorem Ipsum")
      }
    }
  })
